### PR TITLE
Project Management Automation: Check error object for parsed errors

### PR DESCRIPTION
**Pull request by aduth**
Previously: #20011

This pull request seeks to fix the check to determine whether an error thrown in the creation of the milestone during the `add-milestone` automation task is an error representing a duplicate entity.

Based on related resources, it should be the case that this can be determined using the parsed `errors` property of the error object.

- https://github.com/octokit/rest.js/issues/684
- https://developer.github.com/v3/#client-errors
- https://github.com/octokit/request.js/blob/2a1d768/src/fetch-wrapper.ts#L79-L80

### CODE REVIEW

File: packages/project-management-automation/lib/add-milestone.js [hunk 1]
```diff
@@ -12,6 +12,23 @@ const REFERENCE_DATE = '2019-08-12';
...
+ * false otherwise. The error is expected to be (though not verified explicitly
+ * as) an instance of Octokit's RequestError (`@octokit/request-error`).
+ *
+ * @see https://github.com/octokit/rest.js/issues/684
+ * @see https://developer.github.com/v3/#client-errors
+ * @see https://github.com/octokit/request.js/blob/2a1d768/src/fetch-wrapper.ts#L79-L80
+ *
+ * @param {Error} error Error to test.
+ *
+ * @return {boolean} Whether error is duplicate validation request error.
```
Comment (epiqueras) [inline packages/project-management-automation/lib/add-milestone.js]:
```suggestion
 * @return {boolean} Whether error is a duplicate validation request error.
```

**Review by epiqueras — APPROVED** [review]

**Comment by aduth** [comment]
The failing build does not seem that it should be related, and is of the same sort of intermittent failure that #18773 had originally hoped to address (ZIP failure download). Since it's not related, and I don't think the E2E jobs are particularly relevant for this package, I'm going to merge anyways.

**Comment by aduth** [comment]
The action ran successfully this time:

https://github.com/WordPress/gutenberg/runs/424037572?check_suite_focus=true

You can also see in the history of this pull request that the bot assigned the milestone:

https://github.com/WordPress/gutenberg/pull/20014#event-3003760709

**Comment by epiqueras** [comment]
Nice work!

**Comment by aduth** [comment]
Thanks for the timely reviews!

Definitely need to think about how to improve the workflow to changes (verifying correctness) in these automations.

**Comment by epiqueras** [comment]
I think actions have a testing framework now.

**Pull request merged**